crates/virtual-node/src/lib.rs

```diff
@@ -54,8 +53,22 @@ lazy_static! {
 /// in order to recursively render the node and all of its children.
 ///
 /// TODO: Make all of these fields private and create accessor methods
+/// TODO: Create a builder to create instances of VirtualNode::Element with
+/// props and children without having to explicitly create a VirtualNodeElement
+#[derive(Debug, PartialEq)]
+pub enum VirtualNode {
+    /// An element node (node type `ELEMENT_NODE`).
+    Element(VirtualNodeElement),
```

Comment:
The naming `VirtualNode` vs `VirtualNodeElement` vs `VirtualNodeText` makes it kind of hard to tell what is what. On paper they all sound like siblings.

I get that in the real DOM there's the same problem where the naming isn't indicative of what is an upcast vs. a downcast..

Maybe `VirtualNodeElement -> VElement` and `VirtualNodeText -> VText` in order to better illustrate that they are lower in the hierarchy?

Reply:
Ah, I like that! Makes things easier to read. I'll adjust the code.